HikVisionSource/src/main/java/io/vantiq/extsrc/HikVisionSource/HikVision.java

```diff
@@ -0,0 +1,1460 @@
+             * struThermometryAlarm.struPtzInfo.fPan,
+             * struThermometryAlarm.struPtzInfo.fTilt,
+             * struThermometryAlarm.struPtzInfo.fZoom, struThermometryAlarm.byAlarmLevel,
+             * struThermometryAlarm.byAlarmType, struThermometryAlarm.byAlarmRule,
+             * struThermometryAlarm.byRuleCalibType, szRegionInfo,
+             * struThermometryAlarm.dwPicLen, struThermometryAlarm.dwThermalPicLen,
+             * struThermometryAlarm.dwThermalInfoLen);
+             */
+        }
+
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("HHmmssSSS");
+        String time = dtf.format(LocalDateTime.now());
+
+        if ((struThermometryAlarm.dwPicLen != 0) && (struThermometryAlarm.pPicBuff != Pointer.NULL)) {
+            String str = String.format("Device_ID_Pic_[%s]_lUerID_[%d]_%s.%s", strIP, pAlarmer.lUserID.intValue(), time,
+                    (struThermometryAlarm.byPicTransType != 0) ? "data" : "jpg");
+            String fullFileName = String.format("%s/%s", DVRImageFolderPath, str);
+
+            DumpToFile(fullFileName, struThermometryAlarm.pPicBuff, struThermometryAlarm.dwPicLen);
+            o.ImageName = str;
+        }
+
+        if ((struThermometryAlarm.dwThermalPicLen != 0) && (struThermometryAlarm.pThermalPicBuff != Pointer.NULL)) {
+            String str = String.format("Device_ID_ThermoPic_[%s]_lUerID_[%d]_%s.%s", strIP, pAlarmer.lUserID.intValue(),
+                    time, (struThermometryAlarm.byPicTransType != 0) ? "data" : "jpg");
+
+            String fullFileName = String.format("%s/%s", DVRImageFolderPath, str);
+
+            DumpToFile(fullFileName, struThermometryAlarm.pThermalPicBuff, struThermometryAlarm.dwThermalPicLen);
+            o.ThermalImageName = str;
+        }
+
+        if (struThermometryAlarm.dwThermalInfoLen > 0 && struThermometryAlarm.pThermalInfoBuff != Pointer.NULL) {
+            String str = String.format("%s/Device_ID_ThermalInfoPic_[%s]_lUerID_[%d]_%s.%s", DVRImageFolderPath, strIP,
+                    pAlarmer.lUserID.intValue(), time, "data");
+
+            DumpToFile(str, struThermometryAlarm.pThermalInfoBuff, struThermometryAlarm.dwThermalInfoLen);
+
+        }
+
+        log.info("Camera {} : {}", camera.CameraId, stringAlarm);
+        o.Extended = struThermometryAlarm;
+
+        sendNotificationWithUpload(o);
+
+    }
+
+    public void uploadThermalImage(ThermalNotification notification, ImageUtil iu) {
+        HikUploadHelper responseHandler = new HikUploadHelper() {
+            @Override
+            public void onSuccess(Object body, okhttp3.Response response) {
+                log.info("Upload Thermal before send notification");
+                oClient.sendNotification(notification);
+
+            }
+
+            @Override
+            public void onError(List<VantiqError> errors, okhttp3.Response response) {
+                // TODO Auto-generated method stub
+
```

Comment:
extra line -- And seems like we should report some error if we get one, shouldn't we?